### ISSUE TITLE
More PSR-12 updates

### ISSUE THREAD

**Issue by Beakerboy** [markrogoyski/math-php#345]
Many files need a blank line between the opening php tag and the namespace declaration. PHPCS can automatically do this.

PHPCS is also warning that if the code is to be run on php7.1 or above, constants need to have visibility declared. I don’t know how you want to handle this one if you want to continue to support 7.0.

**Comment by markrogoyski** [comment]
PSR-2 did not have an opinion on this, and in their examples did not put a space between the opening <?php tag and the namespace declaration.

The PSR-12 standard added language about this, but I'm not sure how to interpret this:
> The header of a PHP file may consist of a number of different blocks. If present, each of the blocks below MUST be separated by a single blank line, and MUST NOT contain a blank line.

It seems like it contradicts itself by saying it _must not contain a blank line_. It's not entirely clear what that means.

I personally like how the code is, but if PSR-12 really requires that I will update everything.

**Comment by markrogoyski** [comment]
The code style should reflect the lowest PHP version that you support. MathPHP v1.0 will support PHP 7.0. I think 2.0 will also support 7.0 (My plan for this is to be some minor interface changes and not a big release). After that I think it will have been long enough to consider upping the minimum version and adding new 7.2 features and retrofitting things that have changed, like constant visibility declarations.

**Comment by Beakerboy** [comment]
A block is a collection of lines, so each block cannot contain a blank line, but different blocks must be separated by a blank line. The opening php tag is considered a block, so there must be a blank line after it.

We might have to add an override on the new visibility sniff to stop  from raising issues.

**Comment by markrogoyski** [comment]
Oh OK. The block itself cannot have a blank line.

**Comment by markrogoyski** [comment]
I've updated all the opening blocks to conform to the new PSR-12 rules.

The PHPCS config for setting the PHP version doesn't seem to have an effect on whether it uses version-appropriate sniffs or not, so I added an exclusion for this sniff, as it does not apply to PHP 7.0: `PSR12.Properties.ConstantVisibility.NotFound`. I'll remove it whenever MathPHP stops supporting PHP 7.0.

I've also updated the build script and Makefile lint target so if PHPCS finds something wrong it will display the name of the sniff to make it easier to look it up.

Thanks again.
